install-deps.sh

```diff
@@ -550,6 +551,22 @@ else
         exit 1
         ;;
     esac
+
+    if $with_ubbd; then
+	    UBBD_TAR="ubbd.tar.gz"
+	    UBBD_VERSION="0.1.1"
+	    UBBD_DOWNLOAD_URL="https://github.com/DataTravelGuide/ubbd/releases/download/v${UBBD_VERSION}/ubbd-${UBBD_VERSION}.tar.gz"
+	    mkdir -p /var/cache/ubbd/
+	    $SUDO curl -sL -o/var/cache/ubbd/${UBBD_TAR} ${UBBD_DOWNLOAD_URL}
+	    RESTORED_DIR=`pwd`
+	    cd /var/cache/ubbd/
+	    tar xzvf ${UBBD_TAR}
+	    cd ubbd-${UBBD_VERSION}/
+	    make libubbd
+	    make libubbd_install
```

Comment:
Thanx for your comment. I am also not sure the best way to integrate ubbd into install-deps.sh.

I want to use package in first version, but I am afraid I cannot support all packages for all version of os and arch of machine to want compile ceph, especially arm package.

So I choose to install libubbd here in install-deps.sh, it is just a simple library and a header file. That compile from the release tarball, and packages are also compiled from release tarball. So that should be working togather.

Again, I am not sure what's the best way to integrate it into install-deps.sh, wish someone can give me better suggestion. thanx

Reply:
Until ubbd packages are widely supported by distributions I would prefer a solution, when if the ceph is built with UBBD, it builds and install the ubbd binary and lib inside the build directory (as all other binaries) and build the ubbd package (similarly how we distribute rbd-nbd). But this is just my opinion, not a request.

Reply:
aha，got it. thanx @trociny